// @prettier/plugin-pug/dist/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.logger = exports.createLogger = exports.Logger = exports.LogLevel = void 0;
var LogLevel;
(function (LogLevel) {
    LogLevel["DEBUG"] = "debug";
    LogLevel["LOG"] = "log";
    LogLevel["INFO"] = "info";
    LogLevel["WARN"] = "warn";
    LogLevel["ERROR"] = "error";
    LogLevel["OFF"] = "off";
})(LogLevel || (exports.LogLevel = LogLevel = {}));
class Logger {
    logger;
    level;
    constructor(logger = console, level = LogLevel.INFO) {
        this.logger = logger;
        this.level = level;
    }
    static isSupportedLogLevel(value) {
        return (typeof value === 'string' &&
            (value === 'debug' ||
                value === 'log' ||
                value === 'info' ||
                value === 'warn' ||
                value === 'error' ||
                value === 'off'));
    }
    setLogLevel(level) {
        this.level = level;
    }
    isDebugEnabled() {
        return this.level <= LogLevel.DEBUG;
    }
    debug(message, ...optionalParams) {
        this.message(LogLevel.DEBUG, message, ...optionalParams);
    }
    log(message, ...optionalParams) {
        this.message(LogLevel.LOG, message, ...optionalParams);
    }
    info(message, ...optionalParams) {
        this.message(LogLevel.INFO, message, ...optionalParams);
    }
    warn(message, ...optionalParams) {
        this.message(LogLevel.WARN, message, ...optionalParams);
    }
    error(message, ...optionalParams) {
        this.message(LogLevel.ERROR, message, ...optionalParams);
    }
    message(level, message, ...optionalParams) {
        if (this.level !== LogLevel.OFF && this.level <= level) {
            if (level !== LogLevel.OFF) {
                this.logger[level](message, ...optionalParams);
            }
        }
    }
}
exports.Logger = Logger;
function createLogger(logger = console) {
    return new Logger(logger);
}
exports.createLogger = createLogger;
exports.logger = createLogger(console);
if (process.env.NODE_ENV === 'test') {
    exports.logger.setLogLevel(LogLevel.DEBUG);
}
let logLevel = process.env.PRETTIER_PLUGIN_PUG_LOG_LEVEL;
if (logLevel) {
    logLevel = logLevel.toLowerCase();
    if (Logger.isSupportedLogLevel(logLevel)) {
        exports.logger.setLogLevel(logLevel);
    }
}
